Adds a class to inject the effective start date

frank-west-iii opened the pull request:
This allows us to use the effective start date to set the low value of
the measure period in every measure. This is used to determine which
patients are used within each measure.

If no effective start date is provided the measures use the original
code which calculates the low date based on the effective date + 60
seconds - 1 year.

If this is used within Ecqm then the reports will no longer create quality report result patient records for patients that fall outside this timeframe. HOWEVER, please note that existing quality report result patient records are never deleted and therefore will still be in the result listing, but not included in any calculation. This will cause confusion and should be considered a bug. We will need to address this in Ecqm.

frank-west-iii committed:
Adds a class to inject the effective start date

This allows us to use the effective start date to set the low value of
the measure period in every measure. This is used to determine which
patients are used within each measure.

If no effective start date is provided the measures use the original
code which calculates the low date based on the effective date + 60
seconds - 1 year.

frank-west-iii commented:
Relevant lines from the hqmf2js code which has these values static and safe to replace or search on.

https://github.com/q-centrix/hqmf2js/blob/master/lib/generator/measure_period.js.erb
https://github.com/q-centrix/hqmf2js/blob/master/lib/generator/execution.rb#L20

dillonwelch commented:
Seems appropriate to me. Is there a straightforward way we can confirm this works with every measure?

Also tagging @rreas for review.

frank-west-iii committed:
Adds effective start date to map reduce functions

This adds the effective start date to all the map reduce functions that
will need to be used in order to calculate the correct results. It also
ensures that the patient cache is tagged with the date so that the cache
cleaner can clear only the relevant cache results.

frank-west-iii commented:
We pushed up a few more changes to this. Although the overall calculation was being done correctly, the finalize function was not taking into consideration the correct date. This is what is being used to calculate provider performances.

This also attaches the effective start date to the patient cache record so that we can use that in the cache cleaner in order to delete only the relevant records.
